src/ui/components/filters/filteroptionscomponent.js

```diff
@@ -44,10 +44,18 @@ class FilterOptionsConfig {
     this.optionType = config.optionType || OptionTypes.STATIC_FILTER;
 
     /**
-     * The list of filter options to display with checked status
+     * The list of filter options to display with checked status as
+     * initially specified in the user configuration
      * @type {object[]}
      */
-    this.options = config.options.map(o => ({ ...o }));
+    this.initialOptions = config.options.map(o => ({ ...o }));
+
+    /**
+     * The list of filter options to display with checked status,
+     * changing with changes in the persistent storage.
+     * @type {object[]}
+     */
+    this.options = Array.from(this.initialOptions);
```

Comment:
Do we want to do a deep copy here as well?

Reply:
We absolutely do, tytyty changed